wordpress-mobile/WordPress-Android
# Post list view toggle fix design issues

Fixes #9530 

Note: This branch is based on https://github.com/wordpress-mobile/WordPress-Android/pull/9765.
Merge instructions

0. Review this PR
1. Review and merge https://github.com/wordpress-mobile/WordPress-Android/pull/9765
2. Update the target branch to `feature/post-filters-compat-view-toggle`
3. Review and merge this PR

Fixes design issues on the post list screen as requested in https://github.com/wordpress-mobile/WordPress-Android/issues/9530#issuecomment-484071724 and fixes design issues for the post list items compact layout

Screenshots https://cloudup.com/isuSw0gKTf9

- Replaced nested relativeLayouts with ContraintLayout and fixed several design issues in https://github.com/wordpress-mobile/WordPress-Android/commit/a6bc51324d8a0bdaa921f31dc250b673cebb2efb, https://github.com/wordpress-mobile/WordPress-Android/pull/9770/commits/3da04bcaf0673fa110288e10fbcafd94397aae35 and https://github.com/wordpress-mobile/WordPress-Android/pull/9770/commits/8eb8ff51ad242ab3f7d9ff1929a75a2ac27c7a13

- `The divider for the compact view should be 1dp` - Fixed in https://github.com/wordpress-mobile/WordPress-Android/commit/76df2112750dc49811bbd1e78f735b5d2175892d

- `use roboto/sans-serif for card labels. Card labels include: timestamps, post state labels (Local Changes, Private, Pending review etc.)` - Fixed in https://github.com/wordpress-mobile/WordPress-Android/commit/c156edc50c31dbbc9d807a9f2812d485fb1693c0

- `Increase padding on the right side of the author selection menu. The value used should be 24dp.` - This is fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/9697

- `Remove "W" list footers. We need to make sure we keep enough space on the base of the list to make sure the FAB is not covering any content as well.` - Fixed in https://github.com/wordpress-mobile/WordPress-Android/commit/79016b89b38e9b1fcda699a3977054c8ca0a57f1,  https://github.com/wordpress-mobile/WordPress-Android/commit/50346549b1c79cf2885ae131a958c4b07059abb8 and https://github.com/wordpress-mobile/WordPress-Android/pull/9770/commits/c6d1e9bb2f12678f3a47e6574f084ea7bbe3b562

- `Adjust the body text line height/spacing to match the reader cards` - https://github.com/wordpress-mobile/WordPress-Android/commit/c156edc50c31dbbc9d807a9f2812d485fb1693c0

- `Copy: Change binned to trashed on "Trash" tab empty state` - Fixed in https://github.com/wordpress-mobile/WordPress-Android/commit/111e4d0aa3ff00da2e3a3c4973e866bf738e7075

- ❓  `Remove bottom margin from progress bar, the progress bar should sit flush to the bottom of the card` - The padding is added there by the system and I can't think of an easy not-hacky solution. 

To test:
1. My Site
2. Post List
3. Change view layout type by tapping on the action in the Toolbar
4. Click on Drafts tab
5. Click on Publish button on one of the items
6. Verify the "uploading item" looks as in the screenshot
7. Scroll to the bottom and make sure the FAB doesn't cover any content
8. Click Trashed tab
9. Either switch to a site with no drafts or delete all the drafts
10. Make sure the empty view says "You don't have any trashed posts"

Update release notes:

- The feature hasn't been released yet.
